enhancements/NNNN-CSEP-template.md

```diff
@@ -0,0 +1,55 @@
+# CSEP-NNNN: Short, descriptive title
+
+<!-- TOC -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats](#notesconstraintscaveats)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed](#infrastructure-needed)
+
+<!-- TOC -->
+
+## Summary
+
+## Motivation
+
+### Goals
+
+### Non-Goals
+
```

Comment:
I have an idea of what this section is about, but I'm not 100% sure. Can we assume that people know the intent of each section, or should there be at least a brief description of each?

Reply:
I think that is fine for now. As said, this process is heavily inspired by the respective process in other OS projects, in particular K8s KEPs. They also have no explanations in their template. What I did to understand what is expected was looking into concrete KEPs that "instantiate" the template. That worked out well for me. So I would suggest to see if that also works out in the CS community.